components/CustomInput/CustomTextInput.jsx

```diff
@@ -0,0 +1,70 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import {TextInput, Text} from 'react-native';
+import {theme} from '../../assets/theme/theme';
+import styles from './style';
+
+
+let delayDebounceFn = null;
+
+const CustomTextInput = ({
+  name,
+  placeholder,
+  onChangeText,
+  onBlur,
+  keyboardType,
+  autoCapitalize,
+  autoCorrect,
+  secureTextEntry,
+  touched,
+  errors,
+}) => {
+
+  const handleInputText = (inputText) => {
+    if (delayDebounceFn) {
+      clearTimeout(delayDebounceFn);
+    }
+
+    delayDebounceFn = setTimeout(() => {
+      onChangeText(inputText);
+      delayDebounceFn = null;
+    }, 500);
+  };
```

Comment:
write useDebounce hook and use here

Reply:
I changed it, added a hook that made it more correct.